### PR TITLE
chore: turn SQL templating off by default

### PR DESCRIPTION
**Pull request by dpgaspar**
### SUMMARY
Use a new feature flag to make it possible to enable/disable SQL templating. 

Note:
Setting is off by default

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [X] Removes existing feature or API

### CODE REVIEW

**Review by willbarrett — APPROVED** [review]

File: UPDATING.md [hunk 1]
```diff
@@ -23,6 +23,8 @@ assists people when migrating to a new version.
 
 ## Next
 
+* [10887](https://github.com/apache/incubator-superset/pull/11172): Breaking change: SQL templating is turned off be default, to turn it by setting `ENABLE_TEMPLATE_PROCESSING` to True on `DEFAULT_FEATURE_FLAGS`
```
Comment (villebro) [inline UPDATING.md:26]: Nit:
```suggestion
* [10887](https://github.com/apache/incubator-superset/pull/11172): Breaking change: SQL templating is turned off be default. To turn it on set `ENABLE_TEMPLATE_PROCESSING` to True on `DEFAULT_FEATURE_FLAGS`
```

**Review by villebro — APPROVED** [review]
Non blocking nit, LGTM.

File: UPDATING.md (hunk 1 above)
Comment (dpgaspar) [reply UPDATING.md]: Thank you!

**Review by bkyryliuk — APPROVED** [review]

File: UPDATING.md [hunk 2]
```diff
@@ -23,6 +23,8 @@ assists people when migrating to a new version.
 
 ## Next
 
+* [10887](https://github.com/apache/incubator-superset/pull/11172): Breaking change: SQL templating is turned off be default. To turn it on set `ENABLE_TEMPLATE_PROCESSING` to True on `DEFAULT_FEATURE_FLAGS`
```
Comment (etr2460) [inline UPDATING.md:26]: while you're in here, could you move the existing `Next` stuff to `0.37.2`? Looks like that wasn't done yet.

Also, make sure you update the PR number

Comment (dpgaspar) [reply UPDATING.md]: Just checked, and the UPDATING is absolutely correct, there are no warning or breaking changes on 0.37.2

Comment (etr2460) [reply UPDATING.md]: ah, gotcha. i forgot that 0.37.2 was only cherrypicked PRs

**Review by etr2460 — APPROVED** [review]

**Comment by willbarrett** [comment]
@dpgaspar I think it's OK to merge this once conflicts are resolved

**Pull request merged**